crates/rolldown/src/hmr/hmr_ast_finalizer.rs

```diff
@@ -316,7 +318,10 @@ impl<'ast> VisitMut<'ast> for HmrAstFinalizer<'_, 'ast> {
               class.id = Some(self.snippet.id("__rolldown_default__", SPAN));
               self.exports.insert("default".into(), "__rolldown_default__".into());
             }
-            *node = ast::Statement::ClassDeclaration(class.take_in(self.alloc));
+            *node = ast::Statement::ClassDeclaration(ArenaBox::new_in(
+              class.as_mut().take_in(self.alloc),
```

Comment:
This is the only difference between the previous rolldown implementation vs the current oxc implementation.